client/allocator/src/freeing_bump.rs

```diff
@@ -404,7 +411,21 @@ impl FreeingBumpHeapAllocator {
 		Header::Occupied(order).write_into(mem, header_ptr)?;
 
 		self.total_size += order.size() + HEADER_SIZE;
-		trace!("Heap size is {} bytes after allocation", self.total_size);
+
+		log::trace!(
+			target: LOG_TARGET,
+			"after allocation, total_size = {}, bumper = {}.",
+			self.total_size,
+			self.bumper,
+		);
+
+		// update trackers if needed.
+		if self.total_size > self.max_total_size {
+			self.max_total_size = self.total_size;
+		}
+		if self.bumper > self.max_bumper {
+			self.max_bumper = self.bumper;
+		}
```

Comment:
Maybe an over the top concern, but is it possible to have all this logic feature gated so we don't have any performance regression?

Reply:
Also I am not totally convinced we need to track `max_bumper` and and `max_total_size`. I would think just logging `total_size` and `bumper` on each alloc and dealloc would be enough

Reply:
Why wouldn't you? you want to know the maximum and don't care about the intermediary values. If you don't track this, for the same execution, you have to stare at your monitor for 20 minutes and go half blind to get a rough idea of what is the maximum. just try it for yourself and enable `trace` logs on my example above :D

Reply:
> Maybe an over the top concern, but is it possible to have all this logic feature gated so we don't have any performance regression?

https://github.com/paritytech/substrate/pull/9291#discussion_r664815735